src/deepcell/data_splitter.py

```diff
@@ -63,12 +70,46 @@ def __init__(self, model_inputs: List[ModelInput], train_transform=None,
         self._use_correlation_projection = use_correlation_projection
         self._centroid_brightness_quantile = centroid_brightness_quantile
         self._centroid_use_mask = centroid_use_mask
+        self._n_depth_bins = n_depth_bins
 
         if center_roi_centroid not in ('test', True, False):
             raise ValueError(f'Invalid value for center_soma. Valid '
                              f'values are "test", True, or False')
         self._center_roi_centroid = center_roi_centroid
 
+        self._create_experiment_labels(experiment_metadatas)
+
+    def _create_experiment_labels(
+            self,
+            experiment_metadatas: List[ExperimentMetadata]):
+        """
+        """
+        self._exp_labels = np.full(len(experiment_metadatas), -1, dtype=int)
+        self._exp_ids = np.empty(len(experiment_metadatas), dtype=np.uint64)
+        depths = np.empty(len(experiment_metadatas), dtype=int)
+
+        for idx, exp_meta in enumerate(experiment_metadatas):
+            if exp_meta['problem_experiment']:
+                self._exp_labels[idx] = self._n_depth_bins
+            elif exp_meta['equipment'].startswith('3P'):
+                self._exp_labels[idx] = self._n_depth_bins + 1
+            self._exp_ids[idx] = int(exp_meta['experiment_id'])
+            depths[idx] = exp_meta['imaging_depth']
+
+        exp_arg_sort = np.argsort(self._exp_ids)
+        self._exp_ids = self._exp_ids[exp_arg_sort]
+        self._exp_labels = self._exp_labels[exp_arg_sort]
+        depths = depths[exp_arg_sort]
+
+        sorted_args = np.argsort(depths)[self._exp_labels < 0]
```

Comment:
Please add doc that the negative labels are those that heven't been assigned yet